exclude panic, can't access excluded file

test on windows:
```
❯ ec -exclude "(build|vs2022|tmp)"                                                                                           
panic: open vs2022/.vs/vAmigaSDL/FileContentIndex/4e84aa57-8be4-4f30-ac08-340db9d4df3b.vsidx: The process cannot access the file because it is being used by another process.
```

```
❯ ec -exclude "(build|vs2022|tmp)" -verbose -debug
...
Don't add vs2022/.vs/vAmigaSDL/CopilotIndices/17.13.430.18484/SemanticSymbols.db-wal to be checked
panic: open vs2022/.vs/vAmigaSDL/FileContentIndex/0018e314-3d0d-4e09-ae3c-b66186939c26.vsidx: The process cannot access the file because it is being used by another process.

goroutine 1 [running]:
github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files.GetContentType({0xc00012c31a, 0x50}, {0x0, 0x0, 0x0, {0xd16e72, 0x1a}, {0x0, 0x0}, 0x1, ...})
        github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files/files.go:182 +0x16e
github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files.AddToFiles({0xc0000a6780, 0x8, 0x8}, {0xc00012c31a, 0x50}, {0x0, 0x0, 0x0, {0xd16e72, 0x1a}, ...})
        github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files/files.go:58 +0x85
github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files.GetFiles({0x0, 0x0, 0x0, {0xd16e72, 0x1a}, {0x0, 0x0}, 0x1, {0x0, 0x0}, ...})
        github.com/editorconfig-checker/editorconfig-checker/v3/pkg/files/files.go:145 +0x2c8
main.main()
        github.com/editorconfig-checker/editorconfig-checker/v3/cmd/editorconfig-checker/main.go:176 +0x39b
```

--- a/pkg/files/files.go
+++ b/pkg/files/files.go
@@ -55,24 +55,28 @@ func IsExcluded(filePath string, config config.Config) (bool, error) {
 // AddToFiles adds a file to a slice if it isn't already in there
 // and meets the requirements and returns the new slice
 func AddToFiles(filePaths []string, filePath string, config config.Config) []string {
+	config.Logger.Debug("AddToFiles: investigating file %s", filePath)
+
+	isExcluded, err := IsExcluded(filePath, config)
+	if err == nil && isExcluded {
+		config.Logger.Verbose("Not adding %s to be checked, it is excluded", filePath)
+		return filePaths
+	}
+
 	contentType, err := GetContentType(filePath, config)
-
-	config.Logger.Debug("AddToFiles: filePath: %s, contentType: %s", filePath, contentType)
-
 	if err != nil {
 		config.Logger.Error("Could not get the ContentType of file: %s", filePath)
 		config.Logger.Error("%v", err.Error())
 	}
+	config.Logger.Debug("AddToFiles: detected ContentType %s on file %s", contentType, filePath)
 
-	isExcluded, err := IsExcluded(filePath, config)
-
-	if err == nil && !isExcluded && IsAllowedContentType(contentType, config) {
-		config.Logger.Verbose("Add %s to be checked", filePath)
-		return append(filePaths, filePath)
+	if err == nil && !IsAllowedContentType(contentType, config) {
+		config.Logger.Verbose("Not adding %s to be checked, it does not have an allowed ContentType", filePath)
+		return filePaths
 	}
 
-	config.Logger.Verbose("Don't add %s to be checked", filePath)
-	return filePaths
+	config.Logger.Verbose("Adding %s to be checked", filePath)
+	return append(filePaths, filePath)
 }
 
 // GetFiles returns all files which should be checked

--- a/pkg/files/files.go
+++ b/pkg/files/files.go
@@ -238,6 +238,10 @@ func GetRelativePath(filePath string) (string, error) {
 // IsAllowedContentType returns whether the contentType is
 // an allowed content type to check or not
 func IsAllowedContentType(contentType string, config config.Config) bool {
+	/*
+		why not use mimetype.EqualsAny:
+		it would only match types exactly, but we allow our users to give an entire type/ category
+	*/
 	for _, allowedContentType := range config.AllowedContentTypes {
 		if strings.Contains(contentType, allowedContentType) {
 			return true

--- a/pkg/files/files.go
+++ b/pkg/files/files.go
@@ -238,11 +238,11 @@ func GetRelativePath(filePath string) (string, error) {
 // IsAllowedContentType returns whether the contentType is
 // an allowed content type to check or not
 func IsAllowedContentType(contentType string, config config.Config) bool {
-	result := false
-
 	for _, allowedContentType := range config.AllowedContentTypes {
-		result = result || strings.Contains(contentType, allowedContentType)
+		if strings.Contains(contentType, allowedContentType) {
+			return true
+		}
 	}
 
-	return result
+	return false
 }

